migrate.go

```diff
@@ -104,62 +104,65 @@ func AddNamedMigration(filename string, up func(*sql.Tx) error, down func(*sql.T
 
 // CollectMigrations returns all the valid looking migration scripts in the
 // migrations folder and go func registry, and key them by version.
+// Passing the zero value for dirpath will result in only migrations
+// registered from AddMigration and AddNamedMigration being collected.
```

Comment:
This looks good. However, as you suggested, I'd prefer if we had two different functions. One of them calling the other.

Reply:
Sure. The only reason I'm kinda concerned is if we look at something like Up, which calls CollectMigrations. How should that case be handled?

Reply:
that's a good point

maybe we go ahead with this then..